app/root.tsx

```diff
@@ -1,8 +1,68 @@
-import { Links, Meta, Outlet, Scripts } from "react-router";
+import { useAtom } from "jotai";
+import { atomWithStorage } from "jotai/utils";
+import {
+  Links,
+  LoaderFunctionArgs,
+  Meta,
+  Outlet,
+  Scripts,
+  useLoaderData,
+} from "react-router";
 import { SideBar } from "~/components/sideBar";
+import { account } from "~/lib/account";
+import { accountCookie } from "~/lib/login";
+import { parseToken } from "~/lib/parseToken";
 import styles from "~/root.module.css";
 
-export function Layout({ children }: { children: React.ReactNode }) {
+export interface LoggedInAccountDatum {
+  id: string;
+  name: string;
+  nickname: string;
+  avatarURL: string;
+  headerURL: string;
+}
+export const loggedInAccountAtom = atomWithStorage<
+  LoggedInAccountDatum | undefined
+>("loggedInAccount", undefined);
+
+export async function loader({
+  request,
+}: LoaderFunctionArgs): Promise<{ id: string; token: string } | false> {
+  const token = await accountCookie.parse(request.headers.get("Cookie"));
+  if (!token) {
+    return false;
+  }
+
+  const parsedToken = parseToken(token);
+  if (parsedToken instanceof Error) {
+    return false;
+  }
+  // ToDo: tokenを返すのはセキュリティー的にやめたほうが良いかもしれない
+  return { id: parsedToken.id, token: token };
+}
+
+export async function Layout({ children }: { children: React.ReactNode }) {
+  const loaderData = useLoaderData<typeof loader>();
+
+  if (loaderData !== false) {
+    const [loggedInAccount, setLoggedInAccount] = useAtom(loggedInAccountAtom);
```

Comment:
```suggestion
  const [loggedInAccount, setLoggedInAccount] = useAtom(loggedInAccountAtom);
  if (loaderData !== false) {
```
Hooks は状況によらず同じ呼び出し順序でないと正常に動作しないはずです. `if` 文の中に `useAtom` を書くとまずいと思います.